packages/common/src/api/tan-query/coins/useArtistCoin.ts

```diff
@@ -1,34 +1,67 @@
 import { Coin } from '@audius/sdk'
-import { useQuery } from '@tanstack/react-query'
+import { QueryClient, useQuery, useQueryClient } from '@tanstack/react-query'
+import { useDispatch } from 'react-redux'
+import { AnyAction, Dispatch } from 'redux'
 
+import { getArtistCoinsBatcher } from '../batchers/getArtistCoinsBatcher'
 import { QUERY_KEYS } from '../queryKeys'
 import { QueryKey, SelectableQueryOptions } from '../types'
 import { useQueryContext } from '../utils'
+import { entityCacheOptions } from '../utils/entityCacheOptions'
 
-export interface UseArtistCoinParams {
-  mint: string
-}
-
-export const getArtistCoinQueryKey = (mint: string) =>
+export const getArtistCoinQueryKey = (mint: string | null | undefined) =>
   [QUERY_KEYS.coin, mint] as unknown as QueryKey<Coin | undefined>
 
+export const getArtistCoinByTickerQueryKey = (ticker: string) =>
+  [QUERY_KEYS.coinByTicker, ticker] as unknown as QueryKey<string>
+
+export const getArtistCoinQueryFn = async (
+  mint: string,
+  queryClient: QueryClient,
+  sdk: any,
+  dispatch: Dispatch<AnyAction>
+) => {
+  const batchGetCoins = getArtistCoinsBatcher({
+    sdk,
+    currentUserId: null,
+    queryClient,
+    dispatch
+  })
+  const coins = await batchGetCoins.fetch(mint)
+  return coins
+}
+
 export const useArtistCoin = <TResult = Coin | undefined>(
-  params: UseArtistCoinParams,
+  mint: string | null | undefined,
```

Comment:
is null or undefined really ok?
why not just optional arg and do null check further down

Reply:
meant to drop, good catch